lib/bypass/instance.ex

```diff
@@ -81,7 +81,7 @@ defmodule Bypass.Instance do
   end
   defp do_handle_call(:down, from, %{socket: socket, ref: ref} = state) when not is_nil(socket) do
     nil = state.caller_awaiting_down  # assertion
-    if state.retained_plug != nil do
+    if state.retained_plugs != [] do
```

Comment:
@alco  If we are doing concurrent requests, then "the specific plug associated with this Bypass instance" doesn't really hold -- there are multiple plugs associated with this instance's `state`.  We need to wait for all of them to finish.

 I think your second statement makes perfect sense to me, and definitely was something I neglected.

Can you please expand on your first statement and explain a little more?